AWS SDK fails to initialize

It seems #4607 has caused an issue with the AWS SDK instrumentation. I guess muzzle may not be parsing classes out of the SPI resource file which is in a different artifact.

I think our javaagent tests don't catch this because of the `implementation` dependency on the library - presumably for safety against these issues we need to change the pattern to something like `compileOnly` and `shadowBundle` so that the test classpath does not have the library included in it. Well I guess https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/conventions/src/main/kotlin/io.opentelemetry.instrumentation.javaagent-testing.gradle.kts#L116 is supposed to be handling it but maybe it doesn't for some reason.

```
[docker-java-stream-1118179615] INFO logback - STDOUT: Caused by: java.lang.ClassNotFoundException: io.opentelemetry.instrumentation.awssdk.v2_2.autoconfigure.TracingExecutionInterceptor
[docker-java-stream-1118179615] INFO logback - STDOUT: 	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(Unknown Source)
[docker-java-stream-1118179615] INFO logback - STDOUT: 	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(Unknown Source)
[docker-java-stream-1118179615] INFO logback - STDOUT: 	at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
[docker-java-stream-1118179615] INFO logback - STDOUT: 	at java.base/java.lang.Class.forName0(Native Method)
[docker-java-stream-1118179615] INFO logback - STDOUT: 	at java.base/java.lang.Class.forName(Unknown Source)
[docker-java-stream-1118179615] INFO logback - STDOUT: 	at software.amazon.awssdk.core.internal.util.ClassLoaderHelper.loadClass(ClassLoaderHelper.java:112)
[docker-java-stream-1118179615] INFO logback - STDOUT: 	at software.amazon.awssdk.core.internal.util.ClassLoaderHelper.loadClass(ClassLoaderHelper.java:71)
[docker-java-stream-1118179615] INFO logback - STDOUT: 	at software.amazon.awssdk.core.interceptor.ClasspathInterceptorChainFactory.createExecutionInterceptor(ClasspathInterceptorChainFactory.java:123)
[docker-java-stream-1118179615] INFO logback - STDOUT: 	... 59 common frames omitted
[docker-java-stream-1118179615] INFO logback - STDOUT: 
```

--- a/conventions/src/main/kotlin/io.opentelemetry.instrumentation.javaagent-shadowing.gradle.kts
+++ b/conventions/src/main/kotlin/io.opentelemetry.instrumentation.javaagent-shadowing.gradle.kts
@@ -6,6 +6,9 @@ plugins {
 
 tasks.withType<ShadowJar>().configureEach {
   mergeServiceFiles()
+  // Merge any AWS SDK service files that may be present (too bad they didn't just use normal
+  // service loader...)
+  mergeServiceFiles("software/amazon/awssdk/global/handlers")
 
   exclude("**/module-info.class")
 

